File: C/primelist.c
/**
 * primelist.c
 *  written by blanclux
 *  This software is distributed on an "AS IS" basis WITHOUT WARRANTY OF ANY KIND.
 */
#include <stdio.h>
#include <stdlib.h>

char   *isprime;

/* Sieve of Eratosthenes */
void
eratosthenes(int n)
{
	int     i, k;

	isprime[0] = 0;
	isprime[1] = 0;

	for (i = 2; i < n; i++) {
		isprime[i] = 1;
	}

	for (k = 2; k * k < n; k++) {
		if (isprime[k]) {
			for (i = k * 2; i < n; i += k) {
				isprime[i] = 0;
			}
		}
	}
}

/*
 * Prime List Generation
 */
int
main(int argc, char *argv[])
{
	int     n = 10000;
	int     i, j = 0;
	int     count = 0;

	if (argc == 2) {
		n = atoi(argv[1]);
	}
	printf("< List of Prime Numbers (by Sieve of Eratosthenes) >\n");
	printf("  Maximum Number : %d\n", n);

	isprime = malloc(n);

	eratosthenes(n);

	for (i = 2; i < n; i++) {
		if (isprime[i]) {
			printf("%3d ", i);
			j++;
			count++;
		}
		if (j != 0 && j % 10 == 0) {
			printf("\n");
			j = 0;
		}
	}
	printf("\n# %d primes.\n", count);

	free(isprime);
	return 0;
}

File: Java/PrimeList.java
/**
 * PrimeList.java
 *  written by blanclux
 *  This software is distributed on an "AS IS" basis WITHOUT WARRANTY OF ANY KIND.
 */
package Blanclux.math;

// Prime List Generation
public class PrimeList {
	static boolean[] isprime;

	public static void main(String[] args) {
		int n = 10000;

		if (args.length != 0) {
			try {
				n = Integer.parseInt(args[0]);
			} catch (Exception e) {
				throw new IllegalArgumentException("Illegal number");
			}
		}
		System.out
				.println("< List of Prime Numbers (by Sieve of Eratosthenes) >");
		System.out.println("  Maximum Number : " + n + "\n");

		isprime = new boolean[n];

		Eratosthenes(n);

		int j = 0;
		int count = 0;
		for (int i = 2; i < n; i++) {
			if (isprime[i]) {
				System.out.print(i + " ");
				j++;
				count++;
			}
			if (j != 0 && j % 10 == 0) {
				System.out.println("");
				j = 0;
			}
		}
		System.out.println("\n# " + count + " primes");
	}

	// Sieve of Eratosthenes
	public static void Eratosthenes(int n) {
		isprime[0] = false;
		isprime[1] = false;

		for (int i = 2; i < n; i++) {
			isprime[i] = true;
		}

		for (int k = 2; k * k < n; k++) {
			if (isprime[k]) {
				for (int i = k * 2; i < n; i += k) {
					isprime[i] = false;
				}
			}
		}
	}
}
